src/main/java/com/google/maps/internal/DayOfWeekAdaptor.java

```diff
@@ -0,0 +1,61 @@
+package com.google.maps.internal;
+
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import com.google.gson.stream.JsonWriter;
+import com.google.maps.model.OpeningHours.Period.OpenClose.DayOfWeek;
+
+import java.io.IOException;
+
+/**
+ * This class handles conversion from JSON to {@link DayOfWeek}.
+ *
+ * <p>Please see
+ * {@url https://google-gson.googlecode.com/svn/trunk/gson/docs/javadocs/com/google/gson/TypeAdapter.html}
+ * for more detail.
+ */
+public class DayOfWeekAdaptor extends TypeAdapter<DayOfWeek> {
+
+  @Override
+  public DayOfWeek read(JsonReader reader) throws IOException {
+    if (reader.peek() == JsonToken.NULL) {
+      reader.nextNull();
+      return null;
+    }
+
+    if (reader.peek() == JsonToken.NUMBER) {
+      int day = reader.nextInt();
+
+      switch (day) {
+        case 0:
+          return DayOfWeek.SUNDAY;
+        case 1:
+          return DayOfWeek.MONDAY;
+        case 2:
+          return DayOfWeek.TUESDAY;
+        case 3:
+          return DayOfWeek.WEDNESDAY;
+        case 4:
+          return DayOfWeek.THURSDAY;
+        case 5:
+          return DayOfWeek.FRIDAY;
+        case 6:
+          return DayOfWeek.SATURDAY;
+        default:
```

Comment:
You could remove this default, and the `} else {` case, and just return `DayOfWeek.UNKNOWN` at the end. But it's a pretty minor nitpick/cleanup.